### PR TITLE
Good bad suite

### PR DESCRIPTION
**Pull request by chaseez**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "command_channels": [
-    1192895746672951340
+    1194820338933043271
```
Comment (byubean) [inline config.json:3]: Don't change the production config for personal testing! Instead, create your own config and use the `--config` argument when launching the bot.

File: config.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
       "engine": "gpt-4-turbo-preview"
     },
     {
-      "name": "gpt-3.5-turbo",
+      "name": "gpt-35-turbo",
```
Comment (byubean) [inline config.json:24]: why?

File: bot_commands.py [hunk 3]
```diff
@@ -11,8 +11,9 @@
 
 
 class BotCommands:
-    def __init__(self, send_message):
+    def __init__(self, send_message, metrics_handler):
         self._send_message = step(send_message)
+        self._metrics_handler = metrics_handler
```
Comment (byubean) [inline bot_commands.py:16]: Why?

**Comment by byubean** [comment]
I suggest the following organization:
- Each test case is a separate folder
  - `conversation.json` has the conversation snippet. It follows the `messages` format expected by OpenAI.
  - `good.json` has a list of good responses
  - `bad.json` has a list of bad responses
  - In `good.json` and `bad.json`, use this format:
    ```json
    [ {'content': 'markdown that we got from OpenAI', 'embedding': [0.4, 0.3, 0.2, ...]}, {'content': 'etc.', ...}, ... ]
    ```

**Comment by byubean** [comment]
Then look through the example conversations. Look for cases where reasonable good and bad responses are possible. For example, look for a conversation where the student pastes in the instructions for the HW assignment; good responses will try to guide the student through solving it; bad responses will tell the student how to solve it, or worse, return code that solves the problem. 

Get several examples like these using different homework assignments. Look for examples in python and in c++. If you need a more recent conversation dump, let me know.

**Comment by byubean** [comment]
Note that the prompt is **not** included in the test suite. The prompt is the *input* to a test, and is therefore not part of the test data.

File: bot_commands.py (hunk 3 above)
Comment (chaseez) [reply bot_commands.py:16]: When I tried testing metrics about a month ago, it wasn't working. I noticed there wasn't a `metrics_handler` being passed in, so I added it in. Do you want me to remove it?

File: my-config.json
Comment (byubean) [inline my-config.json:1]: Don't check in your personal testing config.

File: scratch/suites/.DS_Store
Comment (byubean) [inline scratch/suites/.DS_Store:1]: .gitignore and remove from commit

File: discord_bot.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ def __init__(self, root_save_folder: Path, config: RubberDuckConfig):
         def create_workflow(wtype: str):
             match wtype:
                 case 'command':
-                    return BotCommands(self.send_message)
+                    return BotCommands(self.send_message, MetricsHandler(metrics_folder))
```
Comment (byubean) [inline discord_bot.py:115]: I don't see any changes to `BotCommands` in this PR. Is this change necessary?

**Review by byubean — APPROVED** [review]
